### PR TITLE
Azure support

### PR DESCRIPTION
**Pull request by faiq**
- adds a new `Authenticatable` for Azure
- checks for errors if `Authenticator` wasn't able to be made
- adds test

### CODE REVIEW

File: core/config_test.go [hunk 1]
```diff
@@ -74,23 +74,23 @@ func (s *ConfigSuite) TestConfigBoxStructs() {
 	_, ok := amzn.Box.Auth.(*AmazonAuth)
 	s.Equal(ok, true)
 	env := util.NewEnvironment(os.Environ()...)
-	authenticator := amzn.Box.Auth.ToAuthenticator(env)
+	authenticator, _ := amzn.Box.Auth.ToAuthenticator(env)
```
Comment (hatchan) [inline core/config_test.go]: Probably a good idea to check the error if it is `nil`

File: auth/auth_helper.go [hunk 2]
```diff
@@ -73,6 +80,10 @@ func GetRegistryAuthenticator(opts CheckAccessOptions) (auth.Authenticator, erro
 		return auth.NewAmazonAuth(opts.AwsRegistryID, opts.AwsAccessKey, opts.AwsSecretKey, opts.AwsRegion, opts.AwsStrictAuth), nil
 	}
 
+	if opts.AzureClientID != "" && opts.AzureClientSecret != "" && opts.AzureSubscriptionID != "" && opts.AzureTenantID != "" && opts.AzureResourceGroupName != "" && opts.AzureRegistryName != "" && opts.AzureLoginServer != "" {
```
Comment (hatchan) [inline auth/auth_helper.go]: This is a pretty long line, maybe make it a multiline or define the logic in a separate func (possibly defined on `opts`)

File: core/config_test.go (hunk 1 above)
Comment (faiq) [inline core/config_test.go]: sure! probably should add the azure stuff in the tests too

File: auth/auth_helper.go (hunk 2 above)
Comment (faiq) [inline auth/auth_helper.go]: i don't really think this is that big of a deal, but if you want i can refactor this and the amazon stuff in a different pull request

**Comment by termie** [comment]
:+1:

**Pull request merged**